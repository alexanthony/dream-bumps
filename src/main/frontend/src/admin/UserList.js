import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn
} from 'material-ui/Table';
import RaisedButton from 'material-ui/RaisedButton';
import {Card, CardTitle, CardText} from 'material-ui/Card';
import {Link} from 'react-router';
import Checkbox from 'material-ui/Checkbox';
import EditorModeEdit from 'material-ui/svg-icons/editor/mode-edit';

import {getAllUsers} from '../rootReducer';

const UserList = ({users}) => (

  <Card>
    <CardTitle title="Users"/>
    <CardText>
      <Table selectable={false}>
        <TableHeader adjustForCheckbox={false} displaySelectAll={false}>
          <TableRow>
            <TableHeaderColumn>ID</TableHeaderColumn>
            <TableHeaderColumn>Login</TableHeaderColumn>
            <TableHeaderColumn>First Name</TableHeaderColumn>
            <TableHeaderColumn>Last Name</TableHeaderColumn>
            <TableHeaderColumn>Activated</TableHeaderColumn>
            <TableHeaderColumn></TableHeaderColumn>
          </TableRow>
        </TableHeader>
        <TableBody
          displayRowCheckbox={false}
        >
          {users.map(user => (
            <TableRow key={user.id}>
              <TableRowColumn>{user.id}</TableRowColumn>
              <TableRowColumn>{user.login}</TableRowColumn>
              <TableRowColumn>{user.firstName}</TableRowColumn>
              <TableRowColumn>{user.lastName}</TableRowColumn>
              <TableRowColumn><Checkbox checked={user.activated} /></TableRowColumn>
              <TableRowColumn><RaisedButton icon={<EditorModeEdit />} containerElement={<Link to={`/admin/users/${user.id}`} />} /></TableRowColumn>
            </TableRow>
          ))}
        </TableBody>
      </Table>

    </CardText>
  </Card>
);

UserList.propTypes = {
  users: PropTypes.array
}

const mapStateToProps = state => ({users: getAllUsers(state)});

export {UserList};

export default connect(mapStateToProps)(UserList);
