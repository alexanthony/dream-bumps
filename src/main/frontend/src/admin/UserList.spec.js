import React from 'react';
import { shallow } from 'enzyme';
import { UserList } from './UserList';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const users = [
    {id: 4, login: 'user4', firstName: 'Joe', lastName: 'Bloggs', activated: true},
    {id: 5, login: 'user5', firstName: 'John', lastName: 'Smith', activated: true},
    {id: 7, login: 'user6', firstName: 'Jane', lastName: 'Smith', activated: false},
    {id: 9, login: 'bump9', firstName: 'Jane', lastName: 'Doe', activated: true},
  ]
  const component = shallow(
    <UserList
      users={users}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});
