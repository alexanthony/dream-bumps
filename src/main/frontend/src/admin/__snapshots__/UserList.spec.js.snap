exports[`test Renders properly 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}>
  <CardTitle
    title="Users" />
  <CardText>
    <Table
      allRowsSelected={false}
      fixedFooter={true}
      fixedHeader={true}
      height="inherit"
      multiSelectable={false}
      selectable={false}>
      <TableHeader
        adjustForCheckbox={false}
        displaySelectAll={false}
        enableSelectAll={true}
        selectAllSelected={false}>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableHeaderColumn>
            ID
          </TableHeaderColumn>
          <TableHeaderColumn>
            Login
          </TableHeaderColumn>
          <TableHeaderColumn>
            First Name
          </TableHeaderColumn>
          <TableHeaderColumn>
            Last Name
          </TableHeaderColumn>
          <TableHeaderColumn>
            Activated
          </TableHeaderColumn>
          <TableHeaderColumn />
        </TableRow>
      </TableHeader>
      <TableBody
        allRowsSelected={false}
        deselectOnClickaway={true}
        displayRowCheckbox={false}
        multiSelectable={false}
        preScanRows={true}
        selectable={true}
        style={Object {}}>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            4
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            user4
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Joe
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Bloggs
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <Checkbox
              checked={true}
              disabled={false}
              labelPosition="right" />
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <RaisedButton
              containerElement={
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/users/4" />
              }
              disabled={false}
              fullWidth={false}
              icon={<EditorModeEdit />}
              labelPosition="after"
              primary={false}
              secondary={false} />
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            5
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            user5
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            John
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Smith
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <Checkbox
              checked={true}
              disabled={false}
              labelPosition="right" />
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <RaisedButton
              containerElement={
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/users/5" />
              }
              disabled={false}
              fullWidth={false}
              icon={<EditorModeEdit />}
              labelPosition="after"
              primary={false}
              secondary={false} />
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            7
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            user6
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Jane
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Smith
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <Checkbox
              checked={false}
              disabled={false}
              labelPosition="right" />
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <RaisedButton
              containerElement={
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/users/7" />
              }
              disabled={false}
              fullWidth={false}
              icon={<EditorModeEdit />}
              labelPosition="after"
              primary={false}
              secondary={false} />
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            9
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            bump9
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Jane
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            Doe
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <Checkbox
              checked={true}
              disabled={false}
              labelPosition="right" />
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            <RaisedButton
              containerElement={
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/users/9" />
              }
              disabled={false}
              fullWidth={false}
              icon={<EditorModeEdit />}
              labelPosition="after"
              primary={false}
              secondary={false} />
          </TableRowColumn>
        </TableRow>
      </TableBody>
    </Table>
  </CardText>
</Card>
`;
